src/functions.rs

```diff
@@ -0,0 +1,198 @@
+    unsafe {
+        let mut child_pid = mem::MaybeUninit::uninit();
+        let mut error = ptr::null_mut();
+        let _ = glib_sys::g_spawn_async_with_fds(
+            working_directory.as_ref().to_glib_none().0,
+            argv.to_glib_none().0,
+            envp.to_glib_none().0,
+            flags.to_glib(),
+            child_setup,
+            Box_::into_raw(super_callback0) as *mut _,
+            child_pid.as_mut_ptr(),
+            stdin_fd.as_raw_fd(),
+            stdout_fd.as_raw_fd(),
+            stderr_fd.as_raw_fd(),
+            &mut error,
+        );
+        let child_pid = from_glib(child_pid.assume_init());
+        if error.is_null() {
+            Ok(child_pid)
+        } else {
+            Err(from_glib_full(error))
+        }
+    }
+}
+
+// #[cfg(any(feature = "v2_58", feature = "dox"))]
+// #[cfg(windows)]
+// pub fn spawn_async_with_fds<
+//     P: AsRef<std::path::Path>,
+//     T: AsRawHandle,
+//     U: AsRawHandle,
+//     V: AsRawHandle,
+// >(
+//     working_directory: P,
+//     argv: &[&str],
+//     envp: &[&str],
+//     flags: SpawnFlags,
+//     child_setup: Option<Box_<dyn FnOnce() + 'static>>,
+//     stdin_fd: T,
+//     stdout_fd: U,
+//     stderr_fd: V,
+// ) -> Result<Pid, Error> {
+//     let child_setup_data: Box_<Option<Box_<dyn FnOnce() + 'static>>> = Box_::new(child_setup);
+//     unsafe extern "C" fn child_setup_func<P: AsRef<std::path::Path>>(
+//         user_data: glib_sys::gpointer,
+//     ) {
+//         let callback: Box_<Option<Box_<dyn FnOnce() + 'static>>> =
+//             Box_::from_raw(user_data as *mut _);
+//         let callback = (*callback).expect("cannot get closure...");
+//         callback()
+//     }
+//     let child_setup = if child_setup_data.is_some() {
+//         Some(child_setup_func::<P> as _)
+//     } else {
+//         None
+//     };
+//     let super_callback0: Box_<Option<Box_<dyn FnOnce() + 'static>>> = child_setup_data;
+//     unsafe {
+//         let mut child_pid = mem::MaybeUninit::uninit();
+//         let mut error = ptr::null_mut();
+//         let _ = glib_sys::g_spawn_async_with_fds(
+//             working_directory.as_ref().to_glib_none().0,
+//             argv.to_glib_none().0,
+//             envp.to_glib_none().0,
+//             flags.to_glib(),
+//             child_setup,
+//             Box_::into_raw(super_callback0) as *mut _,
+//             child_pid.as_mut_ptr(),
+//             stdin_fd.as_raw_handle() as usize as _,
+//             stdout_fd.as_raw_handle() as usize as _,
+//             stderr_fd.as_raw_handle() as usize as _,
+//             &mut error,
+//         );
+//         let child_pid = from_glib(child_pid.assume_init());
+//         if error.is_null() {
+//             Ok(child_pid)
+//         } else {
+//             Err(from_glib_full(error))
+//         }
+//     }
+// }
+
+pub fn spawn_async_with_pipes<P: AsRef<std::path::Path>>(
```

Comment:
Probably also not on Windows

Reply:
Not just "probably", same problem. So definitely not on Windows for now :)

Reply:
I disabled the whole file for windows. ;)

Reply:
Ah that also works, but makes the `#[cfg(not(windows))]` further up not required anymore :)